alacritty_terminal/src/grid/storage.rs

```diff
@@ -173,25 +176,41 @@ impl<T> Storage<T> {
         }
     }
 
-    /// Truncate the invisible elements from the raw buffer
+    /// Truncate the invisible elements from the raw buffer.
     pub fn truncate(&mut self) {
         self.inner.rotate_left(self.zero);
         self.inner.truncate(self.len);
 
         self.zero = 0;
     }
 
-    /// Dynamically grow the storage buffer at runtime
+    /// Dynamically grow the storage buffer at runtime. This function tries to minimize
+    /// reallocations by using minimun step for growing internal storage. If you want to avoid this
+    /// behavior use `initialize_exact`.
     pub fn initialize(&mut self, num_rows: usize, template_row: Row<T>)
     where
         T: Clone,
     {
-        let mut new = vec![template_row; num_rows];
+        if self.len + num_rows > self.inner.len() {
+            let len = self.len;
+            self.initialize_exact(max(num_rows, MIN_INIT_SIZE), template_row);
```

Comment:
Using `max` here with `MIN_INIT_SIZE` doesn't make sense to me. This seems like it will always update by `MIN_INIT_SIZE` instead of the requested `num_rows`.

_Related to next comment_

Reply:
No, it's not. num_rows could be larger then `MIN_INIT_SIZE`.